operator/internal/controller/konfluxrbac_controller.go

```diff
@@ -0,0 +1,185 @@
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the KonfluxRBAC object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.19.0/pkg/reconcile
+func (r *KonfluxRBACReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := logf.FromContext(ctx)
+
+	// Fetch the KonfluxRBAC instance
+	konfluxRBAC := &konfluxv1alpha1.KonfluxRBAC{}
+	if err := r.Get(ctx, req.NamespacedName, konfluxRBAC); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	log.Info("Reconciling KonfluxRBAC", "name", konfluxRBAC.Name)
+
+	// Apply all embedded manifests
+	if err := r.applyManifests(ctx, konfluxRBAC); err != nil {
+		log.Error(err, "Failed to apply manifests")
+		SetFailedCondition(konfluxRBAC, RBACConditionTypeReady, "ApplyFailed", err)
+		if updateErr := r.Status().Update(ctx, konfluxRBAC); updateErr != nil {
+			log.Error(updateErr, "Failed to update status")
+		}
+		return ctrl.Result{}, err
+	}
+
+	// Check the status of owned deployments and update KonfluxRBAC status
+	if err := UpdateComponentStatuses(ctx, r.Client, konfluxRBAC, RBACConditionTypeReady); err != nil {
+		log.Error(err, "Failed to update component statuses")
+		SetFailedCondition(konfluxRBAC, RBACConditionTypeReady, "FailedToGetDeploymentStatus", err)
+		if updateErr := r.Status().Update(ctx, konfluxRBAC); updateErr != nil {
+			log.Error(updateErr, "Failed to update status")
+		}
+		return ctrl.Result{}, err
+	}
+
+	log.Info("Successfully reconciled KonfluxRBAC")
+	return ctrl.Result{}, nil
+}
+
+// applyManifests loads and applies all embedded manifests to the cluster.
+func (r *KonfluxRBACReconciler) applyManifests(ctx context.Context, owner *konfluxv1alpha1.KonfluxRBAC) error {
+	log := logf.FromContext(ctx)
+
+	rbacManifest, err := manifests.GetManifest(manifests.RBAC)
+	if err != nil {
+		return fmt.Errorf("failed to get manifests for RBAC: %w", err)
+	}
+
+	objects, err := parseManifests(rbacManifest)
+	if err != nil {
+		return fmt.Errorf("failed to parse manifests for RBAC: %w", err)
+	}
+
+	for _, obj := range objects {
+		// Set ownership labels and owner reference
+		if err := setOwnership(obj, owner, string(manifests.RBAC), r.Scheme); err != nil {
+			return fmt.Errorf("failed to set ownership for %s/%s (%s) from %s: %w",
+				obj.GetNamespace(), obj.GetName(), obj.GetKind(), manifests.RBAC, err)
+		}
+
+		if err := applyObject(ctx, r.Client, obj); err != nil {
+			if obj.GroupVersionKind().Group == "cert-manager.io" || obj.GroupVersionKind().Group == "kyverno.io" {
```

Comment:
I had the same issue in my pr, so I moved this value to a const. You can rebase this one on my prs.